src/Importify/Cabal.hs

```diff
@@ -3,28 +3,102 @@
+                                                        condExecutables, condTreeData,
+                                                        defaultExtensions, exeModules,
+                                                        libBuildInfo, libModules,
+                                                        modulePath, targetBuildDepends)
 import           Distribution.PackageDescription.Parse (readPackageDescription)
 import           Distribution.Verbosity                (normal)
+import           Language.Haskell.Extension            (Extension (..))
+
+type TargetMap = Map String String
+type ExtensionsMap = Map String [String]
 
 readCabal :: FilePath -> IO GenericPackageDescription
 readCabal = readPackageDescription normal
 
--- TODO: also collect for executables and make unique
 -- TODO: what about version bounds?
 getLibs :: GenericPackageDescription -> [String]
-getLibs GenericPackageDescription{..} =
-    maybe []
-          (map dependencyName . targetBuildDepends . libBuildInfo . condTreeData)
-          condLibrary
+getLibs = nub . concat . map (map dependencyName . targetBuildDepends) . getBuildInfos
+
+getBuildInfos :: GenericPackageDescription -> [BuildInfo]
+getBuildInfos GenericPackageDescription{..} =
+    maybe [] ((:[]) . libBuildInfo . condTreeData) condLibrary ++
+    map (buildInfo . condTreeData . snd) condExecutables
+
+getExtensionMaps :: GenericPackageDescription -> (TargetMap, ExtensionsMap)
+getExtensionMaps GenericPackageDescription{..} =
+    ( foldr Map.union Map.empty $ libTargetsMaps ++ exeTargetsMaps
+    , foldr Map.union Map.empty $ libExtensionsMaps ++ exeExtensionsMaps)
+  where
+    (libTargetsMaps, libExtensionsMaps) = unzip $ map (collectLibraryMaps . condTreeData) $ maybeToList condLibrary
+    (exeTargetsMaps, exeExtensionsMaps) = unzip $ do
+        (name, condTree) <- condExecutables
+        pure $ collectExecutableMaps name $ condTreeData condTree
+
+collectLibraryMaps :: Library -> (TargetMap, ExtensionsMap)
+collectLibraryMaps lib = collectModuleMaps "library" (map toFilePath $ libModules lib) (defaultExtensions $ libBuildInfo lib)
+
+collectExecutableMaps :: String -> Executable -> (TargetMap, ExtensionsMap)
+collectExecutableMaps exeName exe = collectModuleMaps ("executable " ++ exeName) (exePath:(map toFilePath $ exeModules exe)) (defaultExtensions $ buildInfo exe)
+  where
+    exePath = take ((length $ modulePath exe) - 3) $ modulePath exe
+
+collectModuleMaps :: String -> [String] -> [Extension] -> (TargetMap, ExtensionsMap)
+collectModuleMaps target mods exts =
+    ( Map.fromList $ zip mods (repeat target)
+    , Map.singleton target (map showExt exts)
+    )
 
 dependencyName :: Dependency -> String
 dependencyName (Dependency PackageName{..} _) = unPackageName
+
+showExt :: Extension -> String
+showExt (EnableExtension ext)   = show ext
+showExt (DisableExtension ext)  = "No" ++ show ext
+showExt (UnknownExtension name) = error "Unknown extension: " ++ name
```

Comment:
Wrong 'prettyExtension`. :) This one is for language-haskell-exts's. Cabal's Extension doesn't have such a function.
Well, since `UnknownExtension` will lead to compilation error I guess it's ok to just pass it to parser/compiler.